batch/models.py

```diff
@@ -82,6 +83,34 @@ def choices(cls):
         return [(key.value, key.name) for key in cls]
 
 
+class JobManager(models.Manager):
+    _STATUS_TRANSITIONS = {
+        (JobStatus.created, JobStatus.running): {"now": "started_on"},
+        (JobStatus.running, JobStatus.done): {"now": "finished_on"},
+        (JobStatus.running, JobStatus.failed): {"now": "finished_on"},
+        (JobStatus.done, JobStatus.collected): {},
+    }
+
+    def update_status(self, *, old: JobStatus, new: JobStatus) -> int:
```

Comment:
Should also contain and id argument.

Reply:
> Should also contain and id argument.

Since querysets can be chained, filtering by ID can be done before updating where needed. This way, we can update status for any given queryset (e.g. for multiple jobs or by PK instead of external ID). Therefore, I suggest leaving `update_status` as a special case of generic `update`.